kafka/integration_test.go

```diff
@@ -1084,3 +1088,364 @@ func TestProducerConsumerHeaders(t *testing.T) {
+		&AdminOptions{ValidateOnly: true})
+	if err != nil {
+		t.Fatalf("CreateTopics(ValidateOnly) failed: %s", err)
+	}
+
+	validateTopicResult(t, result, expError)
+
+	// Now create the topics
+	t.Logf("Creating topics\n")
+	ctx, cancel = context.WithTimeout(context.Background(), maxDuration)
+	defer cancel()
+	result, err = a.CreateTopics(ctx, newTopics, nil)
+	if err != nil {
+		t.Fatalf("CreateTopics() failed: %s", err)
+	}
+
+	validateTopicResult(t, result, expError)
+
+	// Attempt to create the topics again, should all fail.
+	t.Logf("Attempt to re-create topics, should all fail\n")
+	for k := range expError {
+		if expError[k].code == ErrNoError {
+			expError[k] = Error{code: ErrTopicAlreadyExists}
+		}
+	}
+	ctx, cancel = context.WithTimeout(context.Background(), maxDuration)
+	defer cancel()
+	result, err = a.CreateTopics(ctx, newTopics, nil)
+	if err != nil {
+		t.Fatalf("CreateTopics#2() failed: %s", err)
+	}
+
+	validateTopicResult(t, result, expError)
+
+	// Add partitions to some of the topics
+	t.Logf("Create new partitions for a subset of topics\n")
+	newParts := make([]NewPartitions, topicCnt/2)
+	expError = map[string]Error{}
+	for i := 0; i < topicCnt/2; i++ {
+		topic := newTopics[i].Topic
+		newParts[i] = NewPartitions{
+			Topic:         topic,
+			NewTotalCount: newTopics[i].NumPartitions + 3,
+		}
+		if i == 1 {
+			// Invalid partition count (less than current)
+			newParts[i].NewTotalCount = newTopics[i].NumPartitions - 1
+			expError[topic] = Error{code: ErrInvalidPartitions}
+		} else {
+			expError[topic] = Error{}
+		}
+		t.Logf("Creating new partitions for %s: %d -> %d: expecting %v\n",
+			topic, newTopics[i].NumPartitions, newParts[i].NewTotalCount, expError[topic])
+	}
+
+	ctx, cancel = context.WithTimeout(context.Background(), maxDuration)
+	defer cancel()
+	result, err = a.CreatePartitions(ctx, newParts, nil)
+	if err != nil {
+		t.Fatalf("CreatePartitions() failed: %s", err)
+	}
+
+	validateTopicResult(t, result, expError)
+
+	// FIXME: wait for topics to become available in metadata instead
+	time.Sleep(5000 * time.Millisecond)
+
+	// Delete the topics
+	deleteTopics := make([]DeleteTopic, topicCnt)
+	for i := 0; i < topicCnt; i++ {
+		deleteTopics[i] = DeleteTopic{newTopics[i].Topic}
+		if i == topicCnt-1 {
+			expError[deleteTopics[i].Topic] = Error{code: ErrUnknownTopicOrPart}
+		} else {
+			expError[deleteTopics[i].Topic] = Error{}
+		}
+	}
+
+	ctx, cancel = context.WithTimeout(context.Background(), maxDuration)
+	defer cancel()
+	result2, err := a.DeleteTopics(ctx, deleteTopics, nil)
+	if err != nil {
+		t.Fatalf("DeleteTopics() failed: %s", err)
+	}
+
+	validateTopicResult(t, result2, expError)
+}
+
+func validateConfig(t *testing.T, results []ConfigResourceResult, expResults []ConfigResourceResult, checkConfigEntries bool) {
+
+	_, file, line, _ := runtime.Caller(1)
+	caller := fmt.Sprintf("%s:%d", path.Base(file), line)
+
+	if len(results) != len(expResults) {
```

Comment:
Could you do all the following using [reflect.DeepEqual](https://golang.org/pkg/reflect/#DeepEqual)?

Reply:
expResult.Config only contains the config entries we changed, while result.Config contains the full broker/topic config - so we would have to prepopulate expResult.Config and overwrite with our changes in that case, that would be more error prone.